keras_cv/layers/aspp.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any
+from typing import List
+from typing import Mapping
+
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class ASPP(tf.keras.layers.Layer):
+    """Implements the Atrous Spatial Pyramid Pooling.
```

Comment:
"Atrous" is a misnomer. Let's chat live

Reply:
"Atrous" is used in the original paper, but I am open to any suggestion of naming.

Reply:
> Note that concept has existed in past literature under different names, for instance the algorithme a trous, an algorithm for wavelet decomposition (Holschneider et al., 1987; Shensa, 1992).

But I think that if we name it dilated is going to confuse many users.

Reply:
Also if 1 ago we accepted dilated:
https://keras.io/examples/vision/deeplabv3_plus/

Reply:
From the [original paper](https://arxiv.org/abs/1606.00915v2):

> This technique has a long history in signal processing, originally developed for the efficient computation of the wavelet transform in a scheme also known as “algorithme a trous” [15]. We use the term atrous convolution as a shorthand for convolution with upsampled filters

They proceed to use the term multiple times in the same paper, and most know it as "atrous convolution" because of that...